Resources/Computing/PoolComputingElement.py

```diff
@@ -181,11 +184,9 @@ def _getProcessorsForJobs(self, kwargs):
     # If there's a maximum number of processors allowed for the job, use that as maximum,
     # otherwise it will use all the remaining processors
     if 'maxNumberOfProcessors' in kwargs and kwargs['maxNumberOfProcessors']:
-      maxNumberOfProcessors = min(int(kwargs['maxNumberOfProcessors']), availableProcessors)
-    else:
-      maxNumberOfProcessors = availableProcessors
+      requestedProcessors = min(int(kwargs['maxNumberOfProcessors']), availableProcessors)
 
-    return maxNumberOfProcessors
+    return requestedProcessors
```

Comment:
There might be a corner case when this is not set.

Reply:
Which is this case ?